identity-wallet/src/state/reducers/dev_mode.rs

```diff
@@ -38,20 +42,248 @@ lazy_static! {
     );
 }
 
-pub async fn set_dev_mode(state: AppState, action: Action) -> Result<AppState, AppError> {
-    if let Some(enabled) = listen::<SetDevMode>(action).map(|payload| payload.enabled) {
-        return Ok(AppState {
-            dev_mode_enabled: enabled,
-            current_user_prompt: None,
-            ..state
-        });
+const PASSWORD: &str = "sup3rSecr3t";
+
+pub async fn load_dev_profile(state: AppState, action: Action) -> Result<AppState, AppError> {
+    info!("Load dev profile: {:?}", action);
+
+    if let Some(dev_profile) = listen::<DevProfile>(action) {
+        match dev_profile.profile {
+            ProfileType::Ferris => return load_ferris_profile().await,
+            ProfileType::Dragon => return load_dragon_profile(state).await,
+            ProfileType::None => {}
+        }
     }
+
     Ok(state)
 }
 
-pub async fn load_dev_profile(_state: AppState, _action: Action) -> Result<AppState, AppError> {
-    info!("load dev profile");
+pub async fn toggle_dev_settings(mut state: AppState, _action: Action) -> Result<AppState, AppError> {
+    info!("Toggle dev settings");
+
+    if state.dev_profile.is_some() {
+        state.dev_profile = None;
+    } else {
+        state.dev_profile = Some(ProfileType::None);
+    }
+
+    state.current_user_prompt = None;
+
+    Ok(state)
+}
+
+pub async fn login_profile(state: AppState) -> Result<AppState, AppError> {
```

Comment:
I would rename this to `unlock_storage()` if that's the only thing it is doing. I consider "login" to be a multi-step process.